### ISSUE TITLE
ArgumentError (wrong number of arguments (given 6, expected 0))

### ISSUE THREAD

**Issue by mgrassotti** [vivekmarakana/iota.lib.rb#14]
Hi! 
I am having problems fetching transfers.
Running
```
   account.getAccountDetails
```
I get `ArgumentError (wrong number of arguments (given 6, expected 0))`
Any idea on how to solve it?

Debugging the source code of the gem, I saw it could come from 
```
curl.absorb(transactionTrits)
/iota-ruby-1.1.7/lib/iota/utils/utils.rb:78
```

Many thanks

**Comment by vivekmarakana** [comment]
Hi,

Can you provide the stack trace for this error?

**Comment by mgrassotti** [comment]
Hi, it seems intermittent, I couldn't recreate it each time. 
Anyway I could recreate it now:
```
$ rake sync_with_iota
Updating accounts with IOTA...
rake aborted!
ArgumentError: wrong number of arguments (given 6, expected 0)
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/utils/utils.rb:78:in `length'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/utils/utils.rb:78:in `absorb'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/utils/utils.rb:78:in `transactionObject'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/wrappers.rb:21:in `block (2 levels) in getTransactionsObjects'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/wrappers.rb:17:in `each'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/wrappers.rb:17:in `block in getTransactionsObjects'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/api.rb:247:in `sendData'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/transport.rb:36:in `sendBatchedCommand'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/api.rb:94:in `getTrytes'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/wrappers.rb:13:in `getTransactionsObjects'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/wrappers.rb:38:in `block in findTransactionObjects'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/utils/broker.rb:32:in `send'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/transport.rb:5:in `sendCommand'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/api.rb:73:in `findTransactions'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/wrappers.rb:34:in `findTransactionObjects'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/api/wrappers.rb:131:in `bundlesFromAddresses'
.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/models/account.rb:79:in `getAccountDetails'
.../lib/tasks/scheduler.rake:5:in `block (2 levels) in <top (required)>'
```

**Comment by vivekmarakana** [comment]
Can you also paste the code which you're running in rake after censoring private info like seed etc?

**Comment by vivekmarakana** [comment]
Can you mention what parameters you're passing to `getAccountDetails` function?

**Comment by mgrassotti** [comment]
I passed no arguments to `getAccountDetails`, so I guess defaults ones are used.
I call it almost like this: 
```
client = IOTA::Client.new(provider: <node_url>)
IOTA::Models::Account.new(client, <seed>).getAccountDetails
```
I understand it's probably impossible to recreate it without the seed, but I saw this issue on different accounts, after doing some transfers, so I think it could occur also on other accounts. 
I could temporary avoid the exception calling it with `getAccountDetails({}, true, false)`, so I guess it's related to `fetchTransfers`.

**Comment by vivekmarakana** [comment]
`.../.rvm/gems/ruby-2.5.0/gems/iota-ruby-1.1.7/lib/iota/utils/utils.rb:78:in `length'`

I think this might be the issue with the C Curl Extension, I'm fixing it in 1.1.8 which is almost ready and will be released in a couple of days

**Comment by mgrassotti** [comment]
@vivekmarakana thanks so much

**Comment by vivekmarakana** [comment]
@mgrassotti I've released new version 1.1.8 with jruby support. Please update your version and check if this issue is resolved or not.

**Comment by mgrassotti** [comment]
@vivekmarakana it seems to work very well now! Thanks!
